lib/structs/project_details.py

```diff
@@ -2,11 +2,13 @@
 from typing import Optional
 
 
+# pylint: disable=too-many-instance-attributes
 @dataclass
 class ProjectDetails:
     name: str
     email: str
     description: str = ""
+    domain: str = None
```

Comment:
`domain: Optional[str] = None`

Reply:
There's an edit I do in the openstack_project where we need to take a domain id as a required parameter - this is because the openstacksdk doesn't pass the parent id as a domain id so we would need to differentiate between the domains for the projects

Reply:
I have a feature branch for the internal project and external project workflow that I'm working on as well that take domain as a parameter (as the domain is a new field in the project creation action)